src/main/java/com/redhat/devtools/lsp4ij/operations/documentation/LSPDocumentationProvider.java

```diff
@@ -107,16 +132,20 @@ public String generateDoc(@NotNull PsiElement element, @Nullable PsiElement orig
             }
             String s = markupContent
                     .stream()
-                    .map(m -> m.getValue())
+                    .map(MarkupContent::getValue)
                     .collect(Collectors.joining("\n\n"));
-            return styleHtml(editor, RENDERER.render(PARSER.parse(s)));
+            return styleHtml(editor, toHTML(s));
         } finally {
             if (originalElement != null) {
                 originalElement.putUserData(TARGET_OFFSET_KEY, null);
             }
         }
     }
 
+    public @NotNull String toHTML(@NotNull String markdown) {
```

Comment:
I think it should be better to move this method in a MarkdownConverter class and write test on MarkdownConverter.

Perhaps MarkdownConverter  could be used in an another location?

Reply:
done